Eplant/views/eFP/Viewer/index.tsx

```diff
@@ -467,4 +474,60 @@ export default class EFPViewer
       {props.geneticElement?.id}
     </Typography>
   )
+
+  citation = (props: {
+    activeData?: EFPViewerData
+    state?: EFPViewerState
+    gene?: GeneticElement | null
+  }) => {
+    const [xmlData, setXMLData] = useState<string[]>([])
+
+    const viewID = props.activeData?.views.find(
+      (v) => v.id == props.state?.activeView,
+    )?.name
+    const viewXML = props.activeData?.views.find(
+      (v) => v.id == props.state?.activeView,
+    )?.xmlURL
+    useEffect(() => {
+      const xmlLoad = async () => {
+        let xmlString = ''
+        if (viewXML) {
+          try {
+            const response = await fetch(viewXML)
+            const data = await response.text()
+            xmlString = data
+          } catch (error) {
+            console.error('Error fetching xmlData:', error)
+          }
+        }
+
+        // Extract <li> tags
+        if (xmlString !== '') {
+          const parser = new DOMParser()
+          const xmlDoc = parser.parseFromString(xmlString, 'text/xml')
+          const listItems = xmlDoc.querySelectorAll('info li')
```

Comment:
A parent element I think? Not super familiar with XML but this is what it's parsing from: https://bar.utoronto.ca/eplant/data/experiment/efps/AbioticStress/Arabidopsis_thaliana.xml

Reply:
Cool! I was just a little worried about the selector because `querySelectorAll` usually takes a CSS selector as an argument and if info was a class, not an element, it would have been an issue. But probably it wouldn't have worked so I was just curious.